warehouse/query-core/src/main/java/datawave/query/planner/FederatedQueryPlanner.java

```diff
@@ -0,0 +1,315 @@
+
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.log4j.Logger;
+
+import com.google.common.base.Preconditions;
+
+import datawave.query.CloseableIterable;
+import datawave.query.config.ShardQueryConfiguration;
+import datawave.query.exceptions.DatawaveQueryException;
+import datawave.query.model.FieldIndexHole;
+import datawave.query.planner.pushdown.rules.PushDownRule;
+import datawave.query.tables.ScannerFactory;
+import datawave.query.util.MetadataHelper;
+import datawave.util.time.TraceStopwatch;
+import datawave.webservice.common.logging.ThreadConfigurableLogger;
+import datawave.webservice.query.Query;
+import datawave.webservice.query.configuration.GenericQueryConfiguration;
+import datawave.webservice.query.configuration.QueryData;
+
+public class FederatedQueryPlanner extends QueryPlanner {
+
+    private static final Logger log = ThreadConfigurableLogger.getLogger(FederatedQueryPlanner.class);
+    private final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd");
+    private final Calendar calendar = Calendar.getInstance();
+
+    private final ShardQueryConfiguration originalConfig;
+    private final DefaultQueryPlanner originalPlanner;
+    private final Date originalBeginDate;
+    private final Date originalEndDate;
+    private String plannedScript;
+
+    public FederatedQueryPlanner(ShardQueryConfiguration config, DefaultQueryPlanner planner) {
+        this.originalConfig = config;
+        this.originalPlanner = planner;
+        this.originalBeginDate = config.getBeginDate();
+        this.originalEndDate = config.getEndDate();
+        Preconditions.checkNotNull(originalBeginDate, "Configuration begin date must not be null");
+        Preconditions.checkNotNull(originalEndDate, "Configuration end date must not be null");
+    }
+
+    @Override
+    public CloseableIterable<QueryData> process(GenericQueryConfiguration genericConfig, String query, Query settings, ScannerFactory scannerFactory)
+                    throws DatawaveQueryException {
+        // Validate the config type.
+        if (!(genericConfig instanceof ShardQueryConfiguration)) {
+            throw new ClassCastException("Config must be an instance of " + ShardQueryConfiguration.class.getSimpleName());
+        }
+
+        ShardQueryConfiguration config = (ShardQueryConfiguration) genericConfig;
+
+        log.debug("Query originally set to execute against date range " + dateFormat.format(originalBeginDate) + "-" + dateFormat.format(originalEndDate));
+
+        // Get the relevant date ranges.
+        // TODO - Determine if we should pass in fields and datatypes to filter on for this query. Can we do this before calling process() on the query?
```

Comment:
The Query object should have the data types filter.  At this point is has already been passed to the configuration object via the ShardQueryLogic initialize method.

Reply:
Understood, I'll incorporate the datatype filters.